kafka-rest/src/main/java/io/confluent/kafkarest/GroupMetadataObserver.java

```diff
@@ -0,0 +1,275 @@
+        final boolean needPartOfData = offsetOpt.nonEmpty()
+                && countOpt.nonEmpty()
+                && countOpt.get() > 0;
+        if (!needPartOfData) {
+          return result;
+        } else {
+          final scala.collection.Set<Topic> slice =
+                  JavaConverters.asScalaSetConverter(result).asScala()
+                          .slice(offsetOpt.get(), offsetOpt.get() + countOpt.get()).toSet();
+          return JavaConverters.setAsJavaSetConverter(slice).asJava();
+        }
+      } else {
+        return Collections.emptySet();
+      }
+    } else {
+      return Collections.emptySet();
+    }
+  }
+
+  /**
+   * <p>Get consumer group description</p>
+   *
+   * @param groupId - group name
+   * @return description of consumer group
+   *     (all consumed topics with all partition offset information)
+   */
+  public ConsumerGroupSubscription getConsumerGroupInformation(String groupId) throws Exception {
+    return getConsumerGroupInformation(groupId, Option.<String>empty(),
+            Option.<Integer>empty(), Option.<Integer>empty());
+  }
+
+  /**
+   * <p>Get consumer group description</p>
+   *
+   * @param groupId   - group name
+   * @param topic     - topic name for filter - default empty topic name
+   * @param offsetOpt - offset for TopicPartitionEntity
+   *                  collection for each consumer member for paging
+   * @param countOpt  - count of elements TopicPartitionEntity
+   *                  collection for each consumer member for paging
+   * @return description of consumer group
+   */
+  public ConsumerGroupSubscription getConsumerGroupInformation(
+          String groupId,
+          Option<String> topic,
+          Option<Integer> offsetOpt,
+          Option<Integer> countOpt) throws Exception {
+    if (Option.apply(adminClientWrapper).nonEmpty()) {
+      final ConsumerGroupDescription consumerGroupSummary =
+              adminClientWrapper.describeConsumerGroups(Collections.singleton(groupId))
+                      .get(groupId);
+      final Collection<MemberDescription> summaries =
+              consumerGroupSummary.members();
+      if (summaries.size() > 0) {
+        log.debug("Get summary list {}", summaries);
+        try (KafkaConsumer kafkaConsumer = createConsumer(groupId, config)) {
+          final List<ConsumerTopicPartitionDescription> consumerTopicPartitionDescriptions =
+              getConsumerTopicPartitionDescriptions(topic, summaries, kafkaConsumer);
+          final Comparator<ConsumerTopicPartitionDescription> confluentTopicPartitionComparator =
+              Comparator.comparingInt(ConsumerTopicPartitionDescription::getPartitionId);
+          consumerTopicPartitionDescriptions.sort(confluentTopicPartitionComparator);
+          final Node coordinatorNode = consumerGroupSummary.coordinator();
+          return new ConsumerGroupSubscription(
+            getPagedTopicPartitionList(consumerTopicPartitionDescriptions, offsetOpt, countOpt),
```

Comment:
nit: Is the formatting of this expression correct? Should not it be 4 extra-spaces instead of 2 extra-spaces?